File: Ahorn/entities/CustomDashBlock.jl
module VivHelperCustomDashBlock
using ..Ahorn, Maple

@mapdef Entity "VivHelper/CustomDashBlock" CustomDashBlock(x::Integer, y::Integer,
width::Integer=8, height::Integer=8, tiletype::String="3",
blendin::Bool=true, canDash::Bool=true, permanent::Bool=true,
FlagOnBreak::String="", AudioEvent::String="event:/game/general/wall_break_stone"
)

const placements = Ahorn.PlacementDict(
    "Custom Dash Block (Flags, Audio) (Viv's Helper)" => Ahorn.EntityPlacement(
        CustomDashBlock,
        "rectangle",
        Dict{String, Any}(),
        Ahorn.tileEntityFinalizer
    )
)

Ahorn.editingOptions(entity::CustomDashBlock) = Dict{String, Any}(
    "tiletype" => Ahorn.tiletypeEditingOptions(),
    "AudioEvent" => String["event:/game/general/wall_break_dirt", "event:/game/general/wall_break_wood", "event:/game/general/wall_break_ice", "event:/game/general/wall_break_stone"]
)

Ahorn.minimumSize(entity::CustomDashBlock) = 8, 8
Ahorn.resizable(entity::CustomDashBlock) = true, true

Ahorn.selection(entity::CustomDashBlock) = Ahorn.getEntityRectangle(entity)

Ahorn.renderAbs(ctx::Ahorn.Cairo.CairoContext, entity::CustomDashBlock, room::Maple.Room) = Ahorn.drawTileEntity(ctx, room, entity)

end

File: Loenn/entities/CustomDashBlock.lua
local fakeTilesHelper = require("helpers.fake_tiles")

local dashBlock = {}

dashBlock.name = "VivHelper/CustomDashBlock"
dashBlock.depth = 0
dashBlock.placements = {
    name = "main",
    data = {
        tiletype = "3",
        blendin = true,
        canDash = true,
        permanent = true,
        width = 8,
        height = 8
    }, fieldInformation = {
        AudioEvent = {options = {"event:/game/general/wall_break_dirt", "event:/game/general/wall_break_wood", "event:/game/general/wall_break_ice", "event:/game/general/wall_break_stone"}}
    }
}

dashBlock.sprite = fakeTilesHelper.getEntitySpriteFunction("tiletype", "blendin")
dashBlock.fieldInformation = fakeTilesHelper.getFieldInformation("tiletype")

return dashBlock